From d40d45b8a047c08d5673ef01056d4b40ccb5e443 Mon Sep 17 00:00:00 2001
From: Xu Jia <xujia39@huawei.com>
Date: Tue, 26 Jul 2022 17:38:38 +0800
Subject: [PATCH] ipv6: fix kabi for ip6_rt_gc_expire in struct netns_ipv6
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: d40d45b8a047c08d5673ef01056d4b40ccb5e443
Modified-by-SEL: Yes, refreshed due to different context


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5ISAH

--------------------------------

Making ip6_rt_gc_expire atomic breaks the KABI of struct netns_ipv6.
This patch uses KABI_REPLACE to fix it.

Signed-off-by: Xu Jia <xujia39@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Reviewed-by: Yue Haibing <yuehaibing@huawei.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/net/netns/ipv6.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/net/netns/ipv6.h
+++ b/include/net/netns/ipv6.h
@@ -79,7 +79,7 @@ struct netns_ipv6 {
 	struct dst_ops		ip6_dst_ops;
 	rwlock_t		fib6_walker_lock;
 	spinlock_t		fib6_gc_lock;
-	atomic_t		ip6_rt_gc_expire;
+	KABI_REPLACE(unsigned int ip6_rt_gc_expire, atomic_t ip6_rt_gc_expire)
 	unsigned long		ip6_rt_last_gc;
 	unsigned char		flowlabel_has_excl;
 #ifdef CONFIG_IPV6_MULTIPLE_TABLES
